Added Google to SAML 2.0 docs

BrJan opened the pull request:
## Description & motivation
<!---
Describe your changes, and why you're making them. Is this linked to an open
issue, a pull request on dbt core, etc?
-->

## To-do before merge
<!---
(Optional -- remove this section if not needed)
Include any notes about things that need to happen before this PR is merged, e.g.:
- [ ] Change the base branch
- [ ] Ensure PR #56 is merged
-->

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [ ] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

## Checklist
If you added new pages (delete if not applicable):
- [ ] The page has been added to `website/sidebars.js`
- [ ] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [ ] The page has been removed from `website/sidebars.js`
- [ ] An entry has been added to `_redirects`

BrJan committed:
Added Google to SAML 2.0 docs

drewbanin commented:
hey @BrJan - thanks for adding these docs! Have you tested this out, or is this coming right from the existing SAML/Okta docs?

IIRC, there's one big problem with Google's approach to SAML -- it doesn't share group membership information! dbt Cloud uses these group memberships to [assign users to dbt Cloud groups](https://docs.getdbt.com/docs/dbt-cloud/access-control/access-control-overview#sso-mappings) in order to assign permissions to users.

We can take this offline if it makes more sense, but this lack of support for IdP group membership claims has been a blocker to us trying out the SAML integration on top of Google to-date. I'm a-ok with merging this PR, but I do think we'll need to add some additional caveats if we do :)

drewbanin commented on website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md:
```diff
@@ -224,6 +224,67 @@ the integration between Okta and dbt Cloud.
+
+### Finish setup
+
+After creating the Okta application, follow the instructions in the [dbt Cloud Setup](#dbt-cloud-setup)
```
```suggestion
After creating the Google application, follow the instructions in the [dbt Cloud Setup](#dbt-cloud-setup)
```

barberscott commented:
@drewbanin context here is that both of the known Google SAML integrations we have with Enterprise plan customers have ended up needing assistance from Support because the instructions we provide are somewhat unclear and Google uses slightly different terminology.  

@BrJan I agree with @drewbanin that the lack of Group expression is an important caveat, *especially* because it prevents license type mapping. Perhaps something along the lines of:

_While it is possible to integrate with Google Workspace via SAML, because Google's current SAML implementation does not support expressing group membership information, we strongly recommend the use of our native [Google Workspace integration](https://docs.getdbt.com/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-google-gsuite)._

_Integrations with Google SAML will be subject to the following limitations:_

- _All users will be given a Developer license type.  Users who are configured with the Read-Only license type will be reverted to a Developer license type on their next login._
- _Group membership in dbt Cloud must be manually managed._

BrJan committed:
Update website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md

Co-authored-by: Drew Banin <drew@fishtownanalytics.com>

BrJan committed:
Update setting-up-sso-with-saml-2.0.md

BrJan commented:
@drewbanin and @barberscott - thanks for your feedback! Apologies I didn't check this earlier. I added the caveats Scott noted. Thoughts? Anything else we should add?

drewbanin commented:
i just removed myself from review. @barberscott you've got the final words on review here!

barberscott commented on website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md:
```diff
@@ -262,3 +323,9 @@ Users in your IdP will now be able to log into the application by navigating to
 access control](/access-control/enterprise-permissions) for your account.
+
+**Please note the following for Google integrations -**
+* While it is possible to integrate with Google Workspace via SAML, because Google's current SAML implementation does not support expressing group membership information, we strongly recommend the use of our native Google Workspace integration.
```
Check the formatting here -- I think we want to NOT have lines 328-329 with bullets, but have lines 330 and 331 with bullets.

barberscott replied:
```suggestion
Because Google's current SAML implementation does not support sending group membership information as part of the SAML response, we strongly recommend the use of [our native Google Workspace integration](/dbt-cloud-enterprise/setting-up-sso-with-google-gsuite).
```

barberscott commented on website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md:
```diff
@@ -262,3 +323,9 @@ Users in your IdP will now be able to log into the application by navigating to
 After configuring an identity provider, you will be able to set up [role-based
 access control](/access-control/enterprise-permissions) for your account.
+
+**Please note the following for Google integrations -**
```
Minor formatting change...

**Please note the following for Google integrations:**

barberscott replied:
```suggestion
**Please note the following for Google Workplace integration via SAML:**
```

barberscott reviewed (changes requested):
A couple of minor changes and then you should good here.

BrJan reviewed (commented):
Looks good to me @barberscott!

barberscott commented on website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md:
```diff
@@ -262,3 +323,9 @@ Users in your IdP will now be able to log into the application by navigating to
+
+**Please note the following for Google integrations -**
+* While it is possible to integrate with Google Workspace via SAML, because Google's current SAML implementation does not support expressing group membership information, we strongly recommend the use of our native Google Workspace integration.
+* Integrations with Google SAML will be subject to the following limitations:
```
```suggestion
Integrations with Google Workspace via SAML will have the following limitations:
```

barberscott reviewed (dismissed):
Couple more changes just to clean up language and formatting.

BrJan committed:
Update website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md

Co-authored-by: Scott Barber <74067474+barberscott@users.noreply.github.com>

BrJan committed:
Update website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md

Co-authored-by: Scott Barber <74067474+barberscott@users.noreply.github.com>

BrJan committed:
Update website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md

Co-authored-by: Scott Barber <74067474+barberscott@users.noreply.github.com>

runleonarun committed:
Merge branch 'current' into BrJan-patch-1

runleonarun committed:
Merge branch 'current' into BrJan-patch-1

runleonarun committed:
Merge branch 'current' into BrJan-patch-1

mirnawong1 committed:
Merge branch 'current' into BrJan-patch-1

mirnawong1 committed:
Update setting-up-sso-with-saml-2.0.md

removed double asterisks

mirnawong1 commented:
hi @barberscott and @BrJan , thanks for your patience on this and I've made some small format tweaks but it looks like it's good to go if you're happy with it @barberscott. One thing I wanted to raise, will we need to include the emea.dbt.com url for the audience URI? or is it [cloud.getdbt.com](http://cloud.getdbt.com/) only

> Audience URI (SP Entity ID): https://cloud.getdbt.com/ AND  https://emea.dbt.com/?

mirnawong1 committed:
Update setting-up-sso-with-saml-2.0.md

removing google workspace bit per scott

mirnawong1 committed:
Merge branch 'current' into BrJan-patch-1

mirnawong1 committed:
Merge branch 'current' into BrJan-patch-1

mirnawong1 commented:
Per meeting w Leona, created issue #1953 created to tackle having a separate page that explains what a hostname means for MT and ST instances in more detail, and can serve as the centralized/uniform page that details `hostname` and its meanings for different regions and accounts like MT/Virtual private. 

Removed the item about Google not sending groups as it's now outdated.

runleonarun commented on website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md:
```diff
@@ -208,6 +208,67 @@ the integration between Okta and dbt Cloud.
+
+### Configure the Google application
+
+Sign into your **Google Admin Console** via an account with super administrator privileges.
```
@mirnawong1 can we add steps for this procedure?

mirnawong1 replied:
done @runleonarun and updated any UI elements that were mentioned too!

runleonarun commented on website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md:
```diff
@@ -208,6 +208,67 @@ the integration between Okta and dbt Cloud.
+
+Sign into your **Google Admin Console** via an account with super administrator privileges.
+
+* From the Admin console Home page, go to **Apps** > **Web and mobile apps**.
```
I'm not sure how `<` or `>` are read by screen readers so I tend to use "and then click" just in case.

mirnawong1 replied:
oh good tip! thank you!

runleonarun commented on website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md:
```diff
@@ -208,6 +208,67 @@ the integration between Okta and dbt Cloud.
+
+| Name           | Name format | Value                | Description                |
+| -------------- | ----------- | -------------------- | -------------------------- |
+| `First name`   | Unspecified | `first_name`         | _The user's first name_    |
```
We should get rid of these italics because they're not necessary.
```suggestion
| `First name`   | Unspecified | `first_name`         | The user's first name.  |
```

runleonarun commented on website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md:
```diff
@@ -208,6 +208,67 @@ the integration between Okta and dbt Cloud.
+| Name           | Name format | Value                | Description                |
+| -------------- | ----------- | -------------------- | -------------------------- |
+| `First name`   | Unspecified | `first_name`         | _The user's first name_    |
+| `Last name`    | Unspecified | `last_name`          | _The user's last name_     |
```
```suggestion
| `Last name`    | Unspecified | `last_name`          | The user's last name.     |
```

runleonarun commented on website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md:
```diff
@@ -208,6 +208,67 @@ the integration between Okta and dbt Cloud.
+| -------------- | ----------- | -------------------- | -------------------------- |
+| `First name`   | Unspecified | `first_name`         | _The user's first name_    |
+| `Last name`    | Unspecified | `last_name`          | _The user's last name_     |
+| `Primary email`| Unspecified | `email`              | _The user's email address_ |
```
```suggestion
| `Primary email`| Unspecified | `email`              |  The user's email address. |
```

runleonarun commented on website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md:
```diff
@@ -208,6 +208,67 @@ the integration between Okta and dbt Cloud.
+
+### Finish Google setup
+
+* From the Admin console Home page, go to **Apps** > **Web and mobile apps**.
```
```suggestion
* From the Admin console Home page, go to **Apps** and then click **Web and mobile apps**.
```

runleonarun commented on website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md:
```diff
@@ -208,6 +208,67 @@ the integration between Okta and dbt Cloud.
+
+Click **Next** to continue.
+
+On the **App Details** page:
```
Maybe we can give them a little direction in the intro sentence:
```suggestion
Make these changes on the App Details page:
```

runleonarun reviewed (approved):
@mirnawong1 I think this can be merged once you add steps and replace `>` with words!

mirnawong1 committed:
Update website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 commented on website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md:
```diff
@@ -208,6 +208,67 @@ the integration between Okta and dbt Cloud.
 
+## Google integration
+
+_Use this section if you are configuring Google as your identity provider_.
```
```suggestion
Use this section if you are configuring Google as your identity provider.
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md

mirnawong1 commented on website/docs/docs/dbt-cloud/dbt-cloud-enterprise/setting-up-sso-with-saml-2.0.md:
```diff
@@ -208,6 +208,67 @@ the integration between Okta and dbt Cloud.
+
+### Configure the Google application
+
+Sign into your **Google Admin Console** via an account with super administrator privileges.
```
```suggestion
1. Sign into your **Google Admin Console** via an account with super administrator privileges.
```

mirnawong1 committed:
Update setting-up-sso-with-saml-2.0.md

added numbered bullets for procedural steps

mirnawong1 committed:
Merge branch 'current' into BrJan-patch-1

mirnawong1 committed:
Update setting-up-sso-with-saml-2.0.md

updated > steps and reworded account settings -> enterprise/sso to scale w new ui

runleonarun committed:
Merge branch 'current' into BrJan-patch-1

mirnawong1 committed:
Merge branch 'current' into BrJan-patch-1

mirnawong1 commented:
approving PR per Leona
